profiler/README.md

```diff
@@ -15,3 +52,84 @@ It should be considered experimental and is completely unsupported.
 |`splunk.profiler.include.agent.internals` | false                  | set to `true` to include agent internal call stacks |
+
+# Escape hatch
+
+The profiler limits its own behavior under two conditions:
+
+* Free disk space is too low (< 100MB free))
+* More than 5 minutes worth of JFR files are backed up on disk
+
+If a recording is already running when this condition is encountered, it will
+be stopped. Later, when the conditions are no longer applicable, then recording will
+be restarted and the profiler will resume operation.
+
+# FAQ / Troubleshooting
+
+### How do I know if it's working?
+
+At startup, the agent will log the string "JFR profiler is active" at `INFO`. You can grep for this in your logs to see
+something like this:
+```
+[otel.javaagent 2021-09-28 18:17:04:246 +0000] [main] INFO com.splunk.opentelemetry.profiler.JfrActivator - JFR profiler is active.
+```
+
+### How can I see the profiler configuration?
+
+The agent will log the profiling configuration at `INFO` during startup. You can grep for the string
+`com.splunk.opentelemetry.profiler.ConfigurationLogger` to see something like this:
+
+```
+[otel.javaagent 2021-09-28 18:17:04:237 +0000] [main] INFO <snip> - -----------------------
+[otel.javaagent 2021-09-28 18:17:04:237 +0000] [main] INFO <snip> - Profiler configuration:
+[otel.javaagent 2021-09-28 18:17:04:238 +0000] [main] INFO <snip> -                 splunk.profiler.enabled : true
+[otel.javaagent 2021-09-28 18:17:04:239 +0000] [main] INFO <snip> -               splunk.profiler.directory : .
+[otel.javaagent 2021-09-28 18:17:04:244 +0000] [main] INFO <snip> -      splunk.profiler.recording.duration : 20s
+[otel.javaagent 2021-09-28 18:17:04:244 +0000] [main] INFO <snip> -              splunk.profiler.keep-files : false
+[otel.javaagent 2021-09-28 18:17:04:245 +0000] [main] INFO <snip> -           splunk.profiler.logs-endpoint : null
+[otel.javaagent 2021-09-28 18:17:04:245 +0000] [main] INFO <snip> -             otel.exporter.otlp.endpoint : http://collector:4317
+[otel.javaagent 2021-09-28 18:17:04:245 +0000] [main] INFO <snip> -            splunk.profiler.tlab.enabled : false
+[otel.javaagent 2021-09-28 18:17:04:246 +0000] [main] INFO <snip> -   splunk.profiler.period.jdk.threaddump : null
+[otel.javaagent 2021-09-28 18:17:04:246 +0000] [main] INFO <snip> - -----------------------
+```
+
+### What about this escape hatch?
+
+If the escape hatch becomes active, it will log with `com.splunk.opentelemetry.profiler.RecordingEscapeHatch`
+(you can grep for this in the logs). You may also look for `"** THIS WILL RESULT IN LOSS OF PROFILING DATA **"`
+as a big hint that things are not well.
+
+You may need to free up some disk space and/or give the JVM more resources.
+
+### What if I'm on an unsupported JVM?
+
+If your JVM does not support JFR, this is detected by the profiler at startup. It will log
+"Java Flight Recorder (JFR) is not available in this JVM. Profiling is disabled." at `WARN`,
```

Comment:
The official name is "JDK Flight Recorder"

Reply:
https://github.com/signalfx/splunk-otel-java/issues/476